lib/src/platform_download.dart

```diff
@@ -0,0 +1,137 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_downloader/flutter_downloader.dart';
+import 'package:flutter_downloader/src/desktop_platform_download.dart';
+import 'package:flutter_downloader/src/download_metadata.dart';
+
+/// A platform specific [Download] which invokes the method channel and make the
+/// download native if the platform has this requirement.
+class PlatformDownload extends DesktopPlatformDownload {
+  /// Create a platform specific [Download].
+  PlatformDownload({
+    required super.baseDir,
+    required super.metadata,
+    required super.id,
+    required int cacheFileSize,
+  }) {
+    if (Platform.isAndroid || Platform.isIOS) {
+      _backChannel = MethodChannel('$_channelId/$id');
+      _backChannel!.setMethodCallHandler(_handlePlatformEvent);
+    }
+    if (metadata.contentLength != null) {
+      progress = (cacheFileSize * 1000) ~/ metadata.contentLength!;
+      if (progress == 1000) {
+        status = DownloadStatus.completed;
+      }
+    }
+  }
+
+  static const _channelId = 'fluttercommunity/flutter_downloader';
+  late final MethodChannel? _backChannel;
+  static const _methodChannel = MethodChannel(_channelId);
+
+  /// Return the local cache directory.
+  static Future<Directory> getLocalDir() async {
+    if (Platform.isAndroid || Platform.isIOS) {
+      final path = await _methodChannel.invokeMethod<String>('getCacheDir');
+      return Directory(path!);
+    } else {
+      return Directory.current;
+    }
+  }
+
+  /// Create a new PlatformDownload from its saved metadata.
+  static Future<PlatformDownload> fromDirectory(
+    String baseDir,
+    String id,
+  ) async =>
```

Comment:
Let's use the block body.

I use arrow body only when the whole function fits into a single line, otherwise it looks ugly, imho.